features/run_matrix.feature

```diff
@@ -0,0 +1,11 @@
+Feature: Run a matrix build
+
+    A user should be able to run a full build matrix (ideally in parallel).
+
+    Scenario: Run a matrix
+        Given a workflow that involves a complicated matrix
+        When I run the workflow
+        Then all builds in the matrix should complete
+        # maybe this too:
+        # And an instance should be launched for each job
+        # And all jobs should run on different instances
```

Comment:
This is kind of inherent to the architecture we seek to design. Not sure if we need to add this.

Reply:
The two "And" statements here are the test that we ran all jobs in parallel. If we want running the matrix in parallel to be a requirement, then we should probably test the requirement. The initial "Then" statement would also be satisfied if the matrix was run serially.

It might be better gherkin to combine the two "And" statements into a single "And all matrix jobs should run in parallel"? There's a trade-off between making the statement represent less code (better for the developer) and making the statement's purpose clearer to readers (better for the client).